### PR TITLE
Feature/quiz question UI

### PR DESCRIPTION
**Pull request by DavidGslade86**
## Purpose: 

Integrate functional quiz component into current working UI

## Summary:

- Created routes for quizes based on quizID 
- Made some suggested changes from previous PR and additionally added functionality changes to provide user feedback after answering questions. 
- linked quiz UI to quiz cards (clicking any quiz card leads to quiz UI with dummy data)
- added conditional rendering to next question and previous question buttons
- added "Submit Quiz" button that is visible after completing a quiz
- added dialogue boxes to give user feedback when skipping a question or submitting a question or quiz 

## Front End Change:

App.jsx
QuestionContent.jsx
QuizContent.jsx
package.json{}

## Test Steps:

From homepage, sign in and click the "Quizzes" button.  Click any quiz card to take you to dummy "addition" quiz.  Feel free to play with answering the questions correctly or incorrectly.  UI should give feedback.  User should be able to submit quiz if user has navigated to the last question or answered all questions.

## Note on styling

Component is not properly styled.  Kat has offered to help with styling as she understands the styling of the app better than I do. Thanks Kat!

### CODE REVIEW

File: src/QuizContent.js [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+            { id: 2, question: "2+3 equals:", options: [3, 4, 5, 6], correctAnswer: 5, userAnswer: null, isCurrentQ:null},
+            { id: 3, question: "2+4 equals:", options: [3, 4, 5, 6], correctAnswer: 6, userAnswer: null, isCurrentQ:null},
+            { id: 4, question: "2+1 equals:", options: [3, 4, 5, 6], correctAnswer: 3, userAnswer: null, isCurrentQ:null}
+        ];
+
+        setQuestions(quizQs);
+        setQuizTitle(title);
+    }, [])
+
+    const handleAnswerSelected = (answer, resetCallback) => {
```
Comment (in43sh) [inline src/QuizContent.js:121]: `handleAnswerSelected` has no parameters but 2 arguments passed

File: src/QuizContent.js [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+        //changes to the next question if more questions exist
+        if(currentQuestionIndex < questions.length) {
+            setCurrentQuestionIndex(currentQuestionIndex+1);
+        }
+
+        resetCallback();
+    };
+
+    const handleSkipQuestion = (resetCallback) => {
```
Comment (in43sh) [inline src/QuizContent.js:136]: `handleSkipQuestion` doesn't have any parameters

File: src/QuizContent.js [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        //changes to the next question if more questions exist
+        if(currentQuestionIndex < questions.length-1) {
+            setCurrentQuestionIndex(currentQuestionIndex+1);
+        }
+
+        resetCallback();
+    };
+
+    const handleBackQuestion = (resetCallback) => {
```
Comment (in43sh) [inline src/QuizContent.js:145]: `handleBackQuestion` doesn't have any parameters. Could you double check if you want to pass any values?

File: src/QuizContent.js [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+                        }
+                    </Typography>
+                    {questions.map((q, index) => {
+                        const correct = q.userAnswer === q.correctAnswer;
+                        const isCurrentQuestion = index === currentQuestionIndex;
+                        console.log(q.userAnswer, q.correctAnswer, correct);
+                        return (
+                            <div key={q.id} style={{
+                                backgroundColor: !q.userAnswer ? 'rgb(211, 209, 209)' : correct ? '#2e7d32' : '#d32f2f',
+                                border: isCurrentQuestion ? '2px solid #1976d2' : 'none',
```
Comment (in43sh) [inline src/QuizContent.js:178]: I would replace actual color values with the ones from MUI.

File: src/QuizContent.js [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+    return (
+        <Container>
+            <Typography variant="h5" align="left" sx = {{pl:2, pb:2}}>{question}</Typography>
+            <RadioGroup
+                value={selected}
+                onChange={(e) => setSelected(e.target.value)}
+                style = {{
+                    paddingLeft:"10px"
+                }}
+                id = "answer-choices"
```
Comment (in43sh) [inline src/QuizContent.js:22]: Please, remove spaces before and after the `=` sign.

File: src/QuizContent.js [hunk 6]
```diff
@@ -0,0 +1,204 @@
+import React, { useState, useEffect } from 'react';
+import {Container, Typography, RadioGroup, FormControlLabel, Radio, Button} from '@mui/material';
+import { Box } from '@mui/system';
+import Grid from '@mui/material/Unstable_Grid2'
+
+
+function QuestionContent({ question, options, onAnswerSelected, onSkipQuestion, onBackQuestion}) {
```
Comment (in43sh) [inline src/QuizContent.js:7]: I would move this to a separate file.

File: src/QuizContent.js [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+        //changes to the next question if more questions exist
+        if(currentQuestionIndex < questions.length && currentQuestionIndex > 0) {
+            setCurrentQuestionIndex(currentQuestionIndex-1);
+        }
+
+        resetCallback();
+    };
+
+    return (
+        <>
```
Comment (in43sh) [inline src/QuizContent.js:155]: You don't need an empty fragment here as you have a child as one Container element.

**Review by in43sh — CHANGES_REQUESTED** [review]
Overall, great work on Quiz ticket. There are some changes in code but there are also some questions about the UI:
1) When you are on the first question, I think it's reasonable disabling the `Previous question` button. Same with the last question and `Next question` button
2) Just to make sure: do you want to be able to submit an answer if the question was not answered?
3) For UX purposes it might be better to click on the whole row box to choose the question not just on the radio button.

File: src/QuizContent.js (hunk 1 above)
Comment (DavidGslade86) [reply src/QuizContent.js]: Hey Sergey, the arguments are being passed inside the QuestionContent component.  The function is just being passed to the component as a prop so you aren't seeing the parameters.

File: src/QuizContent.js (hunk 2 above)
Comment (DavidGslade86) [reply src/QuizContent.js]: Same as above, the reset callback is inside the QuestionContent component

File: src/QuizContent.js (hunk 3 above)
Comment (DavidGslade86) [reply src/QuizContent.js]: Same as above

File: src/QuizContent.js (hunk 5 above)
Comment (DavidGslade86) [reply src/QuizContent.js]: Prettier will do it for me!

File: src/QuizContent.js (hunk 6 above)
Comment (DavidGslade86) [reply src/QuizContent.js]: Done!

File: src/QuizContent.js (hunk 7 above)
Comment (DavidGslade86) [reply src/QuizContent.js]: great, done!

File: src/QuizContent.js [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+    const handleAnswerSelected = (answer, resetCallback) => {
+        const selectedAnswer = parseInt(answer, 10);
+        console.log("Selected answer:", selectedAnswer);
+        const updatedQuestions = [...questions];
+        updatedQuestions[currentQuestionIndex].userAnswer = selectedAnswer;
+        setQuestions(updatedQuestions);
+
+        //changes to the next question if more questions exist
+        if(currentQuestionIndex < questions.length) {
+            setCurrentQuestionIndex(currentQuestionIndex+1);
```
[the remark this answers is not in the document]
Comment (DavidGslade86) [reply src/QuizContent.js]: I like it! I will implement this change

File: src/RegisterForm.js
[the remark this answers is not in the document]
Comment (DavidGslade86) [reply src/RegisterForm.js:1]: It's old, It will not be merged

File: package.json [hunk 9]
```diff
@@ -3,6 +3,10 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@emotion/react": "^11.11.1",
```
[the remark this answers is not in the document]
Comment (DavidGslade86) [reply package.json]: Not to me either.  I'll try to remove them and see if anything breaks.

File: src/App.js [hunk 10]
```diff
@@ -22,7 +25,9 @@ const [message, setMessage] = useState('');
 
   return (
     <>
-      <h1>{message}</h1>
+      {message ? <h1>{message}</h1> : <h1></h1>} 
+      <QuizContent/>
+      {/*<RegisterForm/>*/}
```
[the remark this answers is not in the document]
Comment (DavidGslade86) [reply src/App.js]: It won't be merged, this is an old version of the app.

**Pull request merged**